### PR TITLE
feat: v0.2.0

### PR DESCRIPTION
**Pull request by allemanfredi**
as per title ...

### CODE REVIEW

File: packages/evm/contracts/Hashi.sol [hunk 1]
```diff
@@ -34,61 +34,89 @@
...
+    function checkHashWithThresholdFromOracles(
         uint256 domain,
-        uint256 id
-    ) public view returns (bytes32 hash) {
-        hash = oracleAdapter.getHashFromOracle(domain, id);
+        uint256 id,
+        uint256 threshold,
+        IOracleAdapter[] calldata oracleAdapters
+    ) external view returns (bool) {
+        if (oracleAdapters.length < threshold) return false;
```
Comment (bertani) [inline packages/evm/contracts/Hashi.sol]: given this would be an illegal call, shouldn't it revert rather than return false?

File: packages/evm/contracts/Hashi.sol [hunk 2]
```diff
@@ -34,61 +34,89 @@
 pragma solidity ^0.8.17;
```
Comment (bertani) [inline packages/evm/contracts/Hashi.sol]: I would recommend to require at least 0.8.20

File: packages/evm/contracts/Hashi.sol [hunk 3]
```diff
@@ -34,61 +34,89 @@
...
+                continue;
+            }
+
+            uint256 num = 0;
+            for (uint256 j = 0; j < hashes.length; ) {
+                if (baseHash == hashes[j]) {
+                    unchecked {
+                        ++num;
+                    }
+                    if (num == threshold) return true;
```
Comment (bertani) [inline packages/evm/contracts/Hashi.sol:67]: what if threshold = 0 && oracleAdapters.length > 0? This would fail, but is it the desired behaviour? In other terms, is threshold = 0 considered valid? I can see some use cases would it would. Replacing "==" here with ">" may be a good start, but not sufficient. A unit test covering these edge cases would be good

File: packages/evm/contracts/Hashi.sol (hunk 2 above)
Comment (allemanfredi) [reply packages/evm/contracts/Hashi.sol]: yes! i'll upgrade it with another pr!

File: packages/evm/contracts/Hashi.sol (hunk 1 above)
Comment (allemanfredi) [reply packages/evm/contracts/Hashi.sol]: makes sense!

File: packages/evm/contracts/Hashi.sol (hunk 3 above)
Comment (allemanfredi) [reply packages/evm/contracts/Hashi.sol:67]: Setting the threshold to 0 is the same as setting it to 1 because if all hashes are 0, it wouldn't enter the j-cycle. It would only enter that cycle if at least one oracle reported something, which is equivalent to setting the threshold to 1.

Comment (bertani) [reply packages/evm/contracts/Hashi.sol:76]: the best approach here could be to make it clear in the code that threshold should always be >= 1

**Review by bertani — APPROVED** [review]
LGTM
